cmd2/cmd2.py

```diff
@@ -302,11 +302,20 @@ class EmptyStatement(Exception):
 # Contains data about a disabled command which is used to restore its original functions when the command is enabled
 DisabledCommand = namedtuple('DisabledCommand', ['command_function', 'help_function'])
 
-# Used to restore state after redirection ends
-# redirecting and piping are used to know what needs to be restored
-RedirectionSavedState = utils.namedtuple_with_defaults('RedirectionSavedState',
-                                                       ['redirecting', 'self_stdout', 'sys_stdout',
-                                                        'piping', 'pipe_proc'])
+
+class RedirectionSavedState(object):
+    # Created by each command to store information about their redirection
+    def __init__(self):
```

Comment:
Recommend taking `self_stdout`, `sys_stdout`, and `pipe_proc_reader` as required arguments and then using their values to assign the three ``self.saved_*`` attributes.